src/lib/rest/curl_context.cpp

```diff
@@ -0,0 +1,94 @@
+#include "curl_context.h"
+
+#include <map>
+
+#include <assert.h>
+
+#include "logMsg/logMsg.h"
+#include "logMsg/traceLevels.h"
+
+
+static pthread_mutex_t contexts_mutex = PTHREAD_MUTEX_INITIALIZER;
```

Comment:
I don`t think this struct including a curl handle will fit in a common file for semaphores. Here, nothing more than usual operations over mutex are used, The rest is specific logic for the "curl context" management. If we would want to reuse the "curl-context" in other application/library/... (what I don't think it's crazy), we would have to copy&paste fragments of common/semaphore. That doesn't seem a very modular approach or a good software engineering practice